diff --git a/openparf/ops/dct/src/dct2_fft2.cpp b/openparf/ops/dct/src/dct2_fft2.cpp
index b0e27db..2073d12 100644
--- a/openparf/ops/dct/src/dct2_fft2.cpp
+++ b/openparf/ops/dct/src/dct2_fft2.cpp
@@ -68,7 +68,9 @@ void idct2Forward(at::Tensor x, at::Tensor expkM, at::Tensor expkN,
         OPENPARF_TENSOR_DATA_PTR(expkN, scalar_t), at::get_num_threads());
 
     auto y =
-        torch::fft::irfft2(at::view_as_complex(buf), {M, N});
+        torch::fft::irfft2(
+            at::view_as_complex(buf),
+            c10::optional<at::IntArrayRef>(std::vector<int64_t>{M, N}));
 
     idct2PostprocessCpuLauncher<scalar_t>(
         OPENPARF_TENSOR_DATA_PTR(y, scalar_t),
@@ -101,7 +103,9 @@ void idctIdxstForward(at::Tensor x, at::Tensor expkM, at::Tensor expkN,
         OPENPARF_TENSOR_DATA_PTR(expkN, scalar_t), at::get_num_threads());
 
     auto y =
-        torch::fft::irfft2(at::view_as_complex(buf), {M, N});
+        torch::fft::irfft2(
+            at::view_as_complex(buf),
+            c10::optional<at::IntArrayRef>(std::vector<int64_t>{M, N}));
 
     idctIdxstPostprocessCpuLauncher<scalar_t>(
         OPENPARF_TENSOR_DATA_PTR(y, scalar_t),
@@ -134,7 +138,9 @@ void idxstIdctForward(at::Tensor x, at::Tensor expkM, at::Tensor expkN,
         OPENPARF_TENSOR_DATA_PTR(expkN, scalar_t), at::get_num_threads());
 
     auto y =
-        torch::fft::irfft2(at::view_as_complex(buf), {M, N});
+        torch::fft::irfft2(
+            at::view_as_complex(buf),
+            c10::optional<at::IntArrayRef>(std::vector<int64_t>{M, N}));
 
     idxstIdctPostprocessCpuLauncher<scalar_t>(
         OPENPARF_TENSOR_DATA_PTR(y, scalar_t),
